src/pages/index.js

```diff
@@ -66,59 +91,202 @@ function handleImageClick(card) {
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
+}
+
+let cardSection;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([data, initialCards]) => {
+    console.log(data);
+    userInfo.setUserInfo(data);
+    profilePicture.src = data.avatar;
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: renderCard,
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleAvatarFormSubmit(data) {
+  updateAvatar.renderLoading(true);
   console.log(data);
-  editProfilePopup.close();
+  api
+    .updateProfilePic(data)
+    .then(() => {
+      console.log(data);
+      profilePicture.src = data.link;
+      updateAvatar.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      updateAvatar.renderLoading(false);
+    });
+}
+
+function handleAddLike(cardID) {
+  api
+    .addLike(cardID)
+    .then((data) => {
+      console.log(data);
+      this.setLikeState(data.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleRemoveLike(cardID) {
+  api
+    .removeLike(cardID)
+    .then((data) => {
+      console.log(data);
+      this.setLikeState(data.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleAttemptDelete(cardElement, cardId) {
+  console.log(cardId);
+  confirmDelete.open();
+  confirmDelete.getCardInfo(cardElement, cardId);
+}
+
+// function handleConfirmDelete(cardId) {
+//   confirmDelete.renderLoading(true);
+//   api
+//     .deleteCard(cardId)
+//     .then(() => {
+//       confirmDelete.close();
+//     })
+//     .catch((err) => console.error(err))
+//     .finally(() => {
+//       confirmDelete.renderLoading(false);
+//     });
+// }
+
+function deleteHandler(cardElement, cardId) {
+  this.renderLoading(true);
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      this.close();
+    })
+    .then(() => {
+      cardElement.remove();
+      cardElement = null;
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      this.renderLoading(false);
+    });
 }
 
+//
+// Update Avatar
+//
+
+const updateAvatar = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarFormSubmit,
+  "Save"
+);
+updateAvatar.setEventListeners();
+
+//
+// Confirmation Modal
+//
+
+const confirmDelete = new PopupConfirmation("#confirm-modal", deleteHandler);
+confirmDelete.setEventListeners();
+
 //
 // EVENT LISTENERS
 //
 
 profileEditBtn.addEventListener("click", () => {
-  const data = userInfo.getUserInfo();
-  profileModalName.value = data.name;
-  profileModalTitle.value = data.about;
-  editFormValidator.resetValidation();
-  editProfilePopup.open();
+  api
+    .getUserInfo()
+    .then((data) => {
+      profileModalName.value = data.name;
+      profileModalTitle.value = data.about;
```

Comment:
Please, don't call `api.getUserInfo()` ( It's a waste of time and making the server busy) when you open the profile popup because you have method `getUserInfo` inside class `UserInfo` for getting this data.